### PR TITLE
feat(shared): add CRUD methods for maintaining fxa user and customer relationship

### LINKED ISSUE #6159: Implement DB API for managing Stripe customer / FxA user associations
Once we have a place in the DB to store the association between FxA user and Stripe customer, we need to implement support for auth-server code to manage and query the associations.

Thinking this should just be basic Create / Read / Update / Delete operations per user to manage association with Stripe customer ID.

Two paths for this:

1. follow the FxA tradition of adding stored procedures and fxa-auth-db-mysql HTTP APIs
1. or, add direct SQL queries to a library in fxa-shared that both auth-server and the new gql-server can use

The latter option may require some ops cooperation to ensure auth-server can directly access the DB. But, it's probably the better option looking forward, since this is a new query - and considering FXA-2335 and that we've been considering decommissioning fxa-auth-db-mysql by moving to direct queries in fxa-shared.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2400)
┆Issue Number: FXA-2400

### PR DESCRIPTION
**Pull request by jackiemunroe**
## Because
* We want to be able to maintain the relationships between customers and users

## This pull request

* Establishes CRUD methods to maintain this relationship

## Issue that this pull request solves

closes #6159

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: packages/fxa-shared/test/db/models/auth/helpers.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import Knex from 'knex';
 
 import { setupAuthDatabase } from '../../../../db';
 import { Account } from '../../../../db/models/auth/account';
+import { AccountCustomers } from '../../../../db/models/auth/account-customers';
```
Comment (chenba) [inline packages/fxa-shared/test/db/models/auth/helpers.ts:13]: This is not used.

File: yarn.lock [hunk 2]
```diff
@@ -32978,6 +32978,7 @@ resolve@~1.11.1:
   dependencies:
     atob: ^2.1.2
     decode-uri-component: ^0.2.0
+  checksum: 9bbd2656937ce422f1526026638a91dacba0822f1ab954ac34728c9b4f2960dbfc22a8463af9fbf74f28f3722263444df79b2b5ffa07de07dbcda2ec5a8c7513
```
Comment (chenba) [inline yarn.lock:32981]: Let's back out these yarn.lock changes.

**Review by chenba — APPROVED** [review]
👌

**Comment by chenba** [comment]
Was there a previous discussion on why we'd want the models in shared?

**Comment by jackiemunroe** [comment]
> Was there a previous discussion on why we'd want the models in shared?

Yes. The conversation was mostly in Slack and, I think during one of our stand-ups, but the gist is related in this issue: https://github.com/mozilla/fxa/issues/6338

**Pull request merged**